### PR TITLE
refactor(roles): batch reorderDriveRoles's write via shared primitive — Phase 6 of task board crash-prevention epic

### PR DESCRIPTION
**Pull request by 2witstudios**
## Summary

- `reorderDriveRoles` already had correct lock ordering via `lockDriveRolesInOrder` (`FOR UPDATE ORDER BY id ASC` over every role row in the drive) — the one pre-existing reorder call site that got this right. It still wrote via N sequential per-row `UPDATE`s though.
- Phase 3 of this epic (#2130, merged to master) built `lockedBatchReorder` — a single batched `UPDATE ... FROM (VALUES ...)` statement. This phase wires that primitive into `reorderDriveRoles`, closing the last of the three reorder-primitive consumer phases (alongside Phases 4/5 for task_items and favorites reorder, running in parallel on separate files).
- **`lockDriveRolesInOrder`'s drive-wide lock was preserved, not replaced.** It's still called first, unchanged, doing both jobs it always did: (1) the `existingIds`/`invalidIds` validation that rejects any submitted role id not in the drive, and (2) acquiring the ordered lock over every role row in the drive — the same lock `createDriveRole`/`updateDriveRole`'s `isDefault` switch takes, which is what serializes those operations against `reorderDriveRoles`, not just reorders against each other. Only the trailing per-row `for` loop was replaced with one `lockedBatchReorder` call (scoped via `scopeWhere: eq(driveRoles.driveId, driveId)`, `touchColumns: [driveRoles.updatedAt]`). `lockedBatchReorder` re-acquiring `FOR UPDATE` on rows already locked in the same transaction is a harmless idempotent no-op in Postgres, not a bug — it was not "optimized away" by dropping `lockDriveRolesInOrder`.

## Test plan

- [x] TDD: added RED-first tests proving (a) the write is now a single `tx.execute()` batched statement instead of N `tx.update()` calls, (b) the drive-wide lock (`lockDriveRolesInOrder`) still precedes the batched write, (c) the empty-`roleIds` no-op is preserved. Verified all three fail against the pre-migration N-updates implementation and pass against the new one.
- [x] Existing invalid-role-id rejection test kept passing unchanged.
- [x] `bun run typecheck` — green (full monorepo, 16/16 tasks).
- [x] `bun run --cwd packages/lib vitest run` — green (371 files, 8521 tests, 4 pre-existing DB-only skips).
- [x] `bun run test:unit` — 16 pre-existing failures in `apps/web` (missing local Postgres `test` role + one unrelated midnight-boundary date test), confirmed present identically on this branch without my changes via `git stash`. Nothing in `packages/lib` regressed.

Epic: PageSpace page `j44e35jwzlhr54fbmruk3k4i` ("Task Board Query & Reorder Safety"), Phase 6 of 8.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01UwxVmYuWCdDPfjR4vtRi2D

### CODE REVIEW

File: packages/lib/src/services/drive-role-service.ts [hunk 1]
```diff
@@ -400,14 +403,16 @@ export async function reorderDriveRoles(
...
-        ));
+    const plan = computeReorderPlan(roleIds.map((id, index) => ({ id, position: index })));
+    if (plan.orderedIds.length > 0) {
+      await lockedBatchReorder(tx, {
+        table: driveRoles,
+        idColumn: driveRoles.id,
+        positionColumn: driveRoles.position,
+        scopeWhere: eq(driveRoles.driveId, driveId),
+        plan,
+        touchColumns: [driveRoles.updatedAt],
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply packages/lib/src/services/drive-role-service.ts:414]: Good catch — this is real and confirmed. `lockedBatchReorder`'s `touchColumns` used SQL `now()` (== `transaction_timestamp()`, frozen at this transaction's BEGIN), so if it blocked waiting on the `FOR UPDATE` lock, the wait could outlast a concurrent writer's commit and stamp an older `updatedAt` than theirs.

Since this PR had already merged by the time this review landed, fixed as a fast-follow in the shared primitive (not just this call site, since Phases 4/5 will also call `lockedBatchReorder` with `touchColumns`): #2165 — swaps `now()` for `clock_timestamp()`. Verified against a real Postgres instance that this reproduces the drift (-504ms with `now()` vs +5ms with `clock_timestamp()` for a 500ms in-transaction delay) and added a deterministic regression test. Leaving this thread open for verification.
